src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -51,54 +71,108 @@ export class ResultComponent implements OnInit, OnChanges {
     ) {}
 
     ngOnInit(): void {
-        if (this.result) {
+        // Get results initially if necessary
+        if (this.participation && !this.hasParticipationResults() && this.course && this.participation.exercise) {
+            this.resultService.findResultsForParticipation(this.course.id, this.participation.exercise.id, this.participation.id).subscribe(results => {
+                this.participation.results = results.body!;
+                this.init(); // imitate the behavior implemented in ngOnChanges() after updates the participation results
```

Comment:
I don't understand the comment grammar-wise

Reply:
Just a remnant of a previous implementation. Thanks for the hint!